### PR TITLE
refactor(ci): Avoid using command make directly

### PR DESCRIPTION
**Pull request by federico-sysdig**
**What type of PR is this?**
/kind cleanup

**Any specific area of the project related to this PR?**
/area build
/area CI

**What this PR does / why we need it**:

Simplifies a bit the configure/generate phase of CMake by letting it create the `build` directory.
Avoids invoking the command `make` directly and relying on CMake instead, thus preparing the build to be used with other build systems, such as Ninja.

Ninja could be easily used after this change just by adding the switch `-G Ninja` to the CMake configure/generate statement. For instance:

```
cmake -B build -S . -G Ninja -DCMAKE_BUILD_TYPE=Release -Dparameter=value
```

In a nutshell this PR realizes the following transformation:

Old:
```
mkdir build
pushd build
cmake -Dparameter=value ..
build -j4 my_exe
popd
```
New:
```
cmake -B build -S . -Dparameter=value
cmake --build build -j4 --target my_exe
```

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

I also simplified the download of the updated version of CMake so that it won't require uncompressing the binaries to a temporary directory that needs to be eventually deleted.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by leogr — DISMISSED** [review]

**Comment by leogr** [comment]
ARM tests are failing cc @FedeDP

**Comment by FedeDP** [comment]
Flaky :)

**Review by Andreagit97 — APPROVED** [review]
/approve
